src/Workspaces/Core/Desktop/Workspace/SQLite/SQLitePersistentStorageService.cs

```diff
@@ -19,6 +20,59 @@ internal partial class SQLitePersistentStorageService : AbstractPersistentStorag
 
         private readonly IPersistentStorageFaultInjector _faultInjectorOpt;
 
+        [DllImport("kernel32.dll")]
+        private static extern IntPtr LoadLibrary(string dllToLoad);
+        private static bool _initialized;
+        private static object _gate = new object();
+
+        internal static bool TryInitializeLibraries()
+        {
+            if (_initialized)
+            {
+                return true;
+            }
+                
+            lock (_gate)
+            {
+                if (_initialized)
```

Comment:
please don't do double checked locking.  i still don't trust that .net has a consistent story around reordered writes and whatnot to make this safe.  It's trivial, and no perf problem, to just take the lock and do the check.

Reply:
Oh, this part was just for testing. This is not needed after all.